### PR TITLE
✨ Reusable image container - Fixes #19

### LINKED ISSUE #19: :art: Create an image-container component
# Create a reusable image container component
---

Quick links: [Summary](#summary) | [Description](#description) | [Technical-Analysis](#Technical-Analysis) | [Acceptance-Criteria](#Acceptance-Criteria)

---

### Summary

As a user, I want a few consistent sets of displaying images.

### Description

As seen on the designs (link below), there are different ways in which we visualise images.

We differentiate the following forms:

image

- **Pill**: The pill is an oval rounded shape with the roundings being on the top and bottom (rounding of each corner is 30% of window height)
- **Stacked**: The stacked cuts out two horizontal pills
- **Window**: A window rounds at the top only (rounding is 30% of height)
- **Card**: A card image only rounds slightly (5 to 10px)

These cutouts are provided by a component called an "image-container".

An "image-container" accepts an input of type "ImageConfig", which has the following properties:

```
1. Title 
2. Image source
3. Visualisation (type ImageVisualisation = Enum { Pill = 0, Stacked = 1, Window = 2, Card = 3 })
4. Max-width (optional)
```

```typeScript
interface ImageConfig {
  title: string
  imageSrc: string
  visualisation: ImageVisualisation
  maxWidth?: string
}

export enum ImageVisualisation {
  Pill = 0,
  stacked = 1,
  Window = 2,
  Card = 3
}
```
The type/model should be stored in `libs/models/schema/ui/images` (images is a module/library)

The name of the file containing the interface is image-config.interface.ts and it's exported by its parent module above.

Your assignment is to implement the component to appear as above.

Please take note of all the design features

#### Link to full design:
```
https://xd.adobe.com/view/b98caf27-6549-49e5-8ed8-c5234788df61-ecdf/screen/cc35e19a-24f2-42ce-8803-ee2942839528/
```
### Technical Analysis

This component will be created in libs/elements/layout/images

The name of the component will be "elewa-image-container"

Use CSS clipping to visualise the different image forms - https://css-tricks.com/clipping-masking-css/

### Acceptance Criteria

1. When rendered the component is pixel-perfect compared to several of the designs above.
2. The component is clean and does not have more than 80 lines
3. The component accepts rectangular width/height images and uses clipping masks to achieve the desired image shapes
4. The correct types are defined in "model/ui/images"
6. The component SCSS uses CSS variables for colour, font, ... Connected to the main style.scss
7. The code is isolated to the libs/elements/layout/images package, unless a justified reason is specified.
8. The design is responsive and reacts to screens smaller than
9. The image renders fast and has multiple srcsets

### PR DESCRIPTION
**Pull request by Kpine-MO**
# Description
Screenshot 2023-08-29 at 07 45 30

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Reference the issue related to this PR as shown below. Every issue has it's own number you can find the issue numbers [here](https://github.com/italanta/italanta-apps/issues).

Fixes # (issue) - *Example*: Fixes #10, #20 NOTE: GitHub issue numbers should be prefixed with **#** to aid with issue tracking.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**: (optional)

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

### CODE REVIEW

File: libs/elements/layout/images/src/lib/components/elewa-image-container/elewa-image-container.component.html [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    <div *ngIf="item.visualisation === 0" class="NgContainer">
+        <h3 class="title">{{ item.title }}</h3>      
+        <div class="singleImg singleImgPill">
+        <img
+          [ngClass]="{
+                pill: item.visualisation === 0,
+              }"
+          src="{{ item.imageSrc }}"
+          alt=""
+        />
```
Comment (Beulah-Matt) [inline libs/elements/layout/images/src/lib/components/elewa-image-container/elewa-image-container.component.html:12]: Like the use of ngClass. :+1:

File: libs/elements/layout/images/src/lib/components/elewa-image-container/elewa-image-container.component.scss [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  width: 200px;
+  height: 280px;
+  border-radius: 10px;
+}
+
+// .NgContainer{
+//     display: flex;
+//     justify-content: center;
+//     align-items: center;
+// }
```
Comment (Beulah-Matt) [inline libs/elements/layout/images/src/lib/components/elewa-image-container/elewa-image-container.component.scss:58]: Always remember to remove comments as well as unused code snippets.

**Review by Beulah-Matt — COMMENTED** [review]
Good attempt.
This issue has however been resolved by PR #122

File: libs/elements/layout/images/src/lib/components/elewa-image-container/elewa-image-container.component.html [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      <div class="singleImg singleImgWindow">
+        <img
+          [ngClass]="{
+                window: item.visualisation === 2,
+              }"
+          src="{{ item.imageSrc }}"
+          alt=""
+        />
+      </div>
+    </div>
```
Comment (LemmyMwaura) [inline libs/elements/layout/images/src/lib/components/elewa-image-container/elewa-image-container.component.html:40]: Next time dynamically render this. Only one div should have been used. The check for the type of config should happen in the ts file

**Review by LemmyMwaura — COMMENTED** [review]
Closing this as it's already been merged. Next time update your PR description

**Pull request closed without merging**